### ISSUE TITLE
[Bug]: Stopped working

### ISSUE THREAD

**Issue by gyp5y** [mariusz-ostoja-swierczynski/tech-controllers#98]
### The problem

Few day ago after upgrading HA to 2024.7.3 my Tech controller stopped working. Here's corresponding errors from log file

Логгер: homeassistant.config_entries
Джерело: config_entries.py:586
Вперше відбулося: 23 липня 2024 р. о 12:12:20 (8 випадки)
Востаннє записано: 23 липня 2024 р. о 23:55:39

Error setting up entry Ct392zpid for tech
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 586, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tech/__init__.py", line 61, in async_setup_entry
    await assets.load_subtitles(language_code, Tech(websession, user_id, token))
  File "/config/custom_components/tech/assets.py", line 25, in load_subtitles
    TRANSLATIONS = await api.get_translations(language)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tech/tech.py", line 190, in get_translations
    result = await self.get(path)
             ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tech/tech.py", line 73, in get
    raise TechError(response.status, await response.text())
custom_components.tech.tech.TechError: (400, '{"error":"Bad Request"}')

Upgraded integration to master - still no luck

### What version of Tech Controllers integration has the issue?

master

### What version of Home Assistant Core has the issue?

core-2024.7.3

### What was the last working version of Tech Controller integration?

2.3.1

**Comment by daroga0002** [comment]
I am running same (latest upgraded core of HA) and dont hve such errors. 

What language setup is in Home assistant?

As looks from error it can be fialing on language setup

**Comment by daroga0002** [comment]
I checked behaviour on few languages and it seems not related.

**Comment by gyp5y** [comment]
It's Ukrainian.

**Comment by daroga0002** [comment]
Ok, I was able to replicate issue. 

In fact it is related to language setup. Integration downloading translations from Tech api for example:

https://emodul.eu/api/v1/i18n/pl

in case of Ukrainian setup it is trying to download default one:
https://emodul.eu/api/v1/i18n/uk 
which is giving 400 error.


As per [API docs](https://api-documentation.emodul.eu/) we have following languages:
![image](https://github.com/user-attachments/assets/d7a0f7a5-dc59-4e63-b3a4-26c506bcccb8)

Languages on tech api:
```
'en', 'fr', 'it', 'es', 'nl', 'pl', 'de', 'cs', 'sk', 'hu', 'ro', 'lt', 'et', 'ru', 'si', 'hr'
```


As per comment in code:
https://github.com/mariusz-ostoja-swierczynski/tech-controllers/blob/bc863470e92f7de86b18fa1f00a1f0a05328d51b/custom_components/tech/tech.py#L191

looks Tech introduced some change on API and now they dont threat `en` as default.

I will prepare fix for this.

**Comment by daroga0002** [comment]
@gyp5y you can test my  fork branch https://github.com/daroga0002/tech-controllers/tree/fix/bug-98


I opened a https://github.com/mariusz-ostoja-swierczynski/tech-controllers/pull/102 so when it will be merged then it will become part of this repository

**Comment by gyp5y** [comment]
Just checked your branch - it works!. Thank you

**Comment by anarion80** [comment]
Fixed via #102

**Comment by daroga0002** [comment]
> Just checked your branch - it works!. Thank you

You can now use stable release version:

https://github.com/mariusz-ostoja-swierczynski/tech-controllers/releases/tag/v2.4.0
